app/docker/docker-compose.athens.yml

```diff
@@ -0,0 +1,35 @@
+version: '3.5'
+
+services:
+
+  athens:
+    container_name: dab_athens
+    image: "${DAB_APPS_ATHENS_IMAGE:-gomods/athens}:${DAB_APPS_ATHENS_TAG:-latest}"
+    labels:
+      description: 'Go Modules Proxy'
+      com.centurylinklabs.watchtower.enable: 'true'
+    restart: on-failure
+    expose:
+      - 3000
+    ports:
+      - "${DAB_APPS_ATHENS_PORT:-3000}:3000"
```

Comment:
Why are we mapping this, especially to quite a low and round number port, seems to be asking for port conflicts. Imo `ports` should have the same value as `expose` where it does not specify the mapped port so that docker chooses one that is ensured to be free.

Reply:
yep get your point, I assume that docker-assigned port will change between starts? If I use this app like `GOPROXY=10.0.1.39:1234` as a variable in places, I'd have to change it?

Reply:
Yeah, for this reason the dab image has a tool called `ishmael` that can be used to dynamically [get the addresses](https://github.com/Nekroze/ishmael#get-the-addresses-a-container-exposes) a container's ports can be reached on so your env var setting would instead use a subshell like so before its usage in an entrypoint:

```bash
GOPROXY=$(ishmael address dab_athens | tail -n 1)
```